tests/wpunit/CommentCacheInvalidationTest.php

```diff
@@ -17,11 +18,88 @@ public function testItWorks() {
 		$this->assertTrue( true );
 	}
 
-	// @todo: need to think through all the comment actions
 	// create comment (unapproved)
+	public function testCreateUnapprovedCommentDoesNotEvictCache() {
+
+		$this->assertEmpty( $this->getEvictedCaches() );
+
+		self::factory()->comment->create_object( [
+			'comment_approved' => false
+		] );
+
+		$this->assertEmpty( $this->getEvictedCaches() );
+
+	}
+
 	// create comment (approved)
+	public function testCreateApprovedCommentEvictsCache() {
+
+		$this->assertEmpty( $this->getEvictedCaches() );
+
+		self::factory()->comment->create_object( [
+			'comment_approved' => true,
+			'comment_post_ID' => $this->published_post->ID,
+		] );
+
+		$evicted_caches = $this->getEvictedCaches();
+		$this->assertNotEmpty( $evicted_caches );
+
+		$this->assertEqualSets([
+			'listComment',
+		], $evicted_caches );
+
+	}
+
 	// approve comment
+	public function testTransitionCommentToApprovedEvictsCache() {
+
+		$this->assertEmpty( $this->getEvictedCaches() );
+
+		self::factory()->comment->update_object( $this->unapproved_comment->comment_ID, [
+			'comment_approved' => true
+		] );
+
+		$evicted_caches = $this->getEvictedCaches();
+		$this->assertNotEmpty( $evicted_caches );
+
+		$this->assertEqualSets([
+			'listComment',
```

Comment:
I believe that this should evict the `singleApprovedCommentByDatabaseId` query as well, because it would have been cached when it was queried, so making it unapproved should evict that cache. 

Might need to listen to a different hook, or ad some logic in the transition comment status listener to handle this case.

Reply:
(sorry, this was intended for the next test, `testTransitionCommentToUnapprovedEvictsCache`)

Reply:
ya, so I believe the logic would be something like: 

- when state transitions from `approved` to anything other than `approved` we should be purge caches with that node, not just the lists

Reply:
Similar to: https://github.com/wp-graphql/wp-graphql-smart-cache/blob/main/src/Cache/Invalidation.php#L403-L405